File: c/DSASheet/2_String/62_.c
#include<stdio.h>
#include<string.h>
#define MAX 100
// PYTHON

char countAndSay(int n){
    if(n==1){
        return "1"; 
    }else{
        int i=0;
        char a[MAX]=countAndSay(n-1);
        char new[MAX];

        while(a[i]!='\0'){
            int count=1;
            char a=(str)a[i];
            while(a[i+1]!='\0'&& a[i]==a[i+1]){
                count++;i++;
            }
            char b=(str)count;
            char c=strcat(b,a);
            new=strcat(new,c);
            i++;
        }
        return new;
    }
}

int main(){
    int num;
    printf("\nEnter the number :");
    scanf("%d",&num);
    printf("the answer is : %c ",countAndSay(num));
}

File: DSASheet/String/1_/62_.py
def countAndSay(n):
    if n==1:
        return "1"
    elif(n==2):
        return "11";
    
    else:
        i=0
        a=countAndSay(n-1)
        b=""
        while(i<len(a)):
            count=1
            c=a[i]
            while(i<len(a)-1 and a[i]==a[i+1]):
                count+=1
                i+=1
            b=b+str(count)+c
            i+=1
        return b

print(countAndSay(4))